package/kedro_viz/data_access/repositories/modular_pipelines.py

```diff
@@ -1,246 +1,324 @@
+        """
         self.tree: Dict[str, ModularPipelineNode] = {
             ROOT_MODULAR_PIPELINE_ID: GraphNode.create_modular_pipeline_node(
                 ROOT_MODULAR_PIPELINE_ID
             )
         }
+        self.node_mod_pipeline_map: Dict[
+            str, Set[str]
+        ] = {}  # Updated to map node_id to a set of modular_pipeline_ids
+
+        self.parameters = set()
+
+    def populate_tree(self, pipeline: KedroPipeline):
+        """
+        Add inputs, outputs, and children to each modular
+        pipeline node based on the given Kedro pipeline.
+
+        Args:
+            pipeline (KedroPipeline): The Kedro pipeline to populate the tree from.
+        """
+        namespaces = sorted(
+            {node.namespace for node in pipeline.nodes if node.namespace}
+        )
+
+        modular_pipeline_ids = {
+            id for ns in namespaces for id in self._explode_namespace(ns)
+        }
+
+        for modular_pipeline_id in sorted(modular_pipeline_ids):
+            self.get_or_create_modular_pipeline(modular_pipeline_id)
+
+            sub_pipeline = pipeline.only_nodes_with_namespace(modular_pipeline_id)
+            rest_of_the_pipeline = pipeline - sub_pipeline
+
+            free_inputs = sub_pipeline.inputs()
+            free_outputs = sub_pipeline.outputs() | (
+                rest_of_the_pipeline.inputs() & sub_pipeline.all_outputs()
+            )
+
+            self.add_inputs(modular_pipeline_id, free_inputs)
+            self.add_outputs(modular_pipeline_id, free_outputs)
+            self.add_children(modular_pipeline_id, sub_pipeline.nodes)
+
+    def _explode_namespace(self, nested_namespace: str) -> List[str]:
+        """
+        Expand the nested namespace into its constituent parts.
+
+        Args:
+            nested_namespace (str): The nested namespace to expand.
+
+        Returns:
+            List[str]: A list of expanded namespaces.
+            Example:
+                >>> nested_namespace = "uk.data_processing.internal"
+                >>> modular_pipeline_repo_obj = ModularPipelinesRepository()
+                >>> expanded_namespace =
+                        modular_pipeline_repo_obj._explode_namespace(nested_namespace)
+                >>> expanded_namespace
+                ["uk", "uk.data_processing", "uk.data_processing.internal"]
+        """
+        if not nested_namespace or "." not in nested_namespace:
+            return [nested_namespace] if nested_namespace else []
+        ns_parts = nested_namespace.split(".")
+        return [".".join(ns_parts[: i + 1]) for i in range(len(ns_parts))]
 
     def get_or_create_modular_pipeline(
         self, modular_pipeline_id: str
     ) -> ModularPipelineNode:
-        """Get the modular pipeline node with the given ID from the repository.
-        If it doesn't exist, create the node, add to the repository and return the instance.
+        """
+        Retrieve or create a modular pipeline node.

```

Comment:
Why change the doc string? The behaviour is still the same right? The previous one also seems to provide more explanation.

Reply:
Yes, it is the same. Reverting back this change.